src/components/NurseryWithdrawals/PlantingProgressList.tsx

```diff
@@ -112,7 +114,20 @@ export default function PlantingProgressList({
     return <CircularProgress sx={{ margin: 'auto' }} />;
   }
 
-  const setPlantingCompleted = (complete: boolean) => {
+  const onModalSubmit = () => {
+    setShowWarningModal(false);
+    setPlantingCompleted(false, true);
+  };
+
+  const setPlantingCompleted = (complete: boolean, skipModal?: boolean) => {
+    if (!complete && !skipModal) {
```

Comment:
can we separate out the 2 paths into separate functions? this approach may leave room for errors ending up in infinite loops

Reply:
also, i'm assuming we will have a confirmation modal at some point? (with Chudo's feedback ofc)

Reply:
Do you mean one function for set plating complete and one for undoing planting complete?

Reply:
It may end up being that way - suggestion is to look at it this way.
The setPlantingCompleted function was correct before, it finds the subzones and dispatches a call etc. and uses the passed in boolean value.
The logic to show warning modal should ideally be isolated from this function.

Reply:
Gotcha!